src/spec/dsl.cr

```diff
@@ -230,6 +243,9 @@ module Spec
     start_time = Time.monotonic
 
     at_exit do
+      if randomizer = @@randomizer
+        root_context.randomize(randomizer)
+      end
```

Comment:
maybe extract these three lines into a `maybe_randomize` helper function (all the code in a function should have the same level of abstraction).